internal/installercache/installercache.go

```diff
@@ -138,14 +140,25 @@ func (i *Installers) Get(ctx context.Context, releaseID, releaseIDMirror, pullSe
 	for {
 		select {
 		case <-ctx.Done():
-			return nil, ctx.Err()
+			err := ctx.Err()
+			if err == context.DeadlineExceeded {
+				i.metricsAPI.InstallerCacheGetReleaseTimeout(releaseID)
+			}
+			return nil, err
 		default:
-			release, err := i.get(releaseID, releaseIDMirror, pullSecret, ocRelease, ocpVersion, clusterID)
+			majorMinorVersion, err := ocRelease.GetMajorMinorVersion(i.log, releaseID, releaseIDMirror, pullSecret)
+			if err != nil {
+				i.log.Warnf("unable to get majorMinorVersion to record metric for %s falling back to full URI", releaseID)
+				majorMinorVersion = releaseID
```

Comment:
as discussed releaseID should not be a value. We can set `unknown` or `undfined` or something if we cannot extract the value from the release (which should never happen anyway)